### PR TITLE
Pull forks using search api / add error for rate-limit hit

### PR DESCRIPTION
**Pull request by ChrisMcKee**
Hi,
  Handy tool; just offering up some changes I made to help me dig my way out of a decade or so of forks 🙈
I blindly hit the rate limit a few times as the upstream checks use a fair few requests; so I added the check upfront.

* Changed `getAllRepos` to fetch users forks using search api, sorted by created date
* Add some more detail to readme.
* Update packages
* Add rate limit check with "time remaining till reset" to rate-limit error

Ta

Chris

### CODE REVIEW

File: fork-cleaner.go [hunk 1]
```diff
@@ -63,29 +55,34 @@ func FindAllForks(
 			return forks, fmt.Errorf("failed to get repository: %s: %w", repo.GetFullName(), err)
 		}
 
-		parent := repo.GetParent()
-
-		// get parent's Issues
-		issues, err := getIssues(ctx, client, login, parent)
-		if err != nil {
-			return forks, fmt.Errorf("failed to get repository's issues: %s: %w", parent.GetFullName(), err)
+		if !skipUpstream {
```
Comment (caarlos0) [inline fork-cleaner.go:58]:
```golang
if skipUpstream {
  forks = append(forks, buildDetails(repo, nil, nil, resp.StatusCode))
  continue
}
```

**Review by caarlos0 — COMMENTED** [review]
thanks for the PR, just a small comment

File: fork-cleaner.go (hunk 1 above)
Comment (ChrisMcKee) [reply fork-cleaner.go]: Sorted 👍

**Comment by caarlos0** [comment]
thank you

**Pull request merged**